### PR TITLE
tests: event_manager: Add unit tests

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds unit tests for Event Manager.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@pdunaj, @Qbicz, could you review?

File: tests/subsys/event_manager/src/modules/test_multicontext_handler.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+			zassert_true(ev->val1 < SOURCE_CNT,
+				     "Invalid source ID");
+
+			if (ev->val1 == SOURCE_T1) {
+				zassert_true(isr_received,
+					     "Incorrect event order");
+				t1_received = true;
+			}
+
+			if (ev->val1 == SOURCE_T2) {
```
Comment (Qbicz) [inline tests/subsys/event_manager/src/modules/test_multicontext_handler.c]: `else if`

Comment (Qbicz) [reply tests/subsys/event_manager/src/modules/test_multicontext_handler.c]: also remove empty line:
```
if {
  ...
} else if {
  ...
}
```

File: tests/subsys/event_manager/src/modules/test_basic.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			/* Ignore other test cases, check if proper test_id. */
+			zassert_true(st->test_id < TEST_CNT,
+				     "test_id out of range");
+			break;
+		}
+
+		return false;
+	}
+
+	zassert_true(false, "Received event without subscription");
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply tests/subsys/event_manager/src/modules/test_basic.c]: done

File: tests/subsys/event_manager/src/modules/test_multicontext_handler.c (hunk 1 above)
Comment (MarekPieta) [reply tests/subsys/event_manager/src/modules/test_multicontext_handler.c]: done

File: tests/subsys/event_manager/src/modules/test_multicontext.c [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	send_event(SOURCE_T2, true);
+}
+
+static void start_test(void)
+{
+	k_thread_create(&thread1, thread_stack1,
+			THREAD_STACK_SIZE,
+			(k_thread_entry_t)thread1_fn,
+			NULL, NULL, NULL,
+			-2, 0, K_NO_WAIT);
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply tests/subsys/event_manager/src/modules/test_multicontext.c]: done

**Comment by MarekPieta** [comment]
I made requested changes

File: tests/subsys/event_manager/src/modules/test_multicontext.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+			THREAD_STACK_SIZE,
+			(k_thread_entry_t)thread1_fn,
+			NULL, NULL, NULL,
+			K_PRIO_COOP(1), 0, K_NO_WAIT);
+
+	k_thread_create(&thread2, thread_stack2,
+			THREAD_STACK_SIZE,
+			(k_thread_entry_t)thread2_fn,
+			NULL, NULL, NULL,
+			K_PRIO_COOP(2), 0, K_NO_WAIT);
```
Comment (Qbicz) [inline tests/subsys/event_manager/src/modules/test_multicontext.c]: `THREAD2_PRIORITY`

File: tests/subsys/event_manager/src/modules/test_multicontext.c [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	send_event(SOURCE_T2, true);
+}
+
+static void start_test(void)
+{
+	k_thread_create(&thread1, thread_stack1,
+			THREAD_STACK_SIZE,
+			(k_thread_entry_t)thread1_fn,
+			NULL, NULL, NULL,
+			K_PRIO_COOP(1), 0, K_NO_WAIT);
```
Comment (Qbicz) [inline tests/subsys/event_manager/src/modules/test_multicontext.c]: You named it `THREAD1_PRIORITY` in header

File: tests/subsys/event_manager/src/modules/test_multicontext.c (hunk 4 above)
Comment (MarekPieta) [reply tests/subsys/event_manager/src/modules/test_multicontext.c]: done

File: tests/subsys/event_manager/src/modules/test_multicontext.c (hunk 5 above)
Comment (MarekPieta) [reply tests/subsys/event_manager/src/modules/test_multicontext.c]: done

**Comment by MarekPieta** [comment]
I made requested changes

**Pull request merged**